### DISCUSSION TITLE
Lost my github repo branch

### DISCUSSION THREAD

**Discussion by Vipul-Bhardwaj777** [EddieHubCommunity/support#5404] [General]
I force-pushed a commit on my GitHub repo using ( git push -f --set-upstream origin master) and now except for that commit, the whole repository is gone.
It's empty now, only the new commit is visible

Is there any way to get back my work🥺
I

**Comment by RaheemAmer** [comment]
If you have only pushed a single commit, you can do the following:

<pre>
git checkout master
git reset --hard HEAD~
git push -f
</pre>

This will restore master to the previous commit. If you made more commits, you can replace HEAD~ with any other commits (a SHA1 hash, a branch name, a tag name, etc.).

## Resources:
1. https://stackoverflow.com/questions/52952002/how-to-recover-after-pushing-to-master-by-mistake
2. https://www.git-tower.com/blog/surviving-with-git-videos/
3. https://medium.com/an-engineer-a-reader-a-guy/git-fix-commits-mistakenly-pushed-to-master-branch-a73e62b7885a
4. https://stackoverflow.com/questions/3973994/how-can-i-recover-from-an-erronous-git-push-f-origin-master

Good luck
